Makefile

```diff
@@ -241,3 +260,40 @@ check-manifests: manifests ## Check if 'make manifests' was run.
 .PHONY: ec
 ec: editorconfig ## Run file formatter checks against all project's files.
 	$(EC)
+
+.PHONY: check-images
+check-images: MANAGER_IMAGE:=$(shell grep -I 'newName: ' ./config/manager/kustomization.yaml | awk -F': ' '{print $$2}')
+check-images: MANAGER_TAG:=$(shell grep -I 'newTag: ' ./config/manager/kustomization.yaml | awk -F': ' '{print $$2}')
+check-images: ## Check if images are the same in Makefile and config/manager/kustomization.yaml
+	@if [ "$(MANAGER_IMAGE)" == "" ];then echo "No manager image found" && exit 1;fi
+	@if [ "$(MANAGER_TAG)" == "" ];then echo "No manager tag found" && exit 1;fi
+	@grep -Iq "IMG ?= $(MANAGER_IMAGE):$(MANAGER_TAG)" ./Makefile || (echo "Images differ" && exit 1)
+
+# A valid oadp-operator git repo fork (for example https://github.com/openshift/oadp-operator)
+OADP_FORK ?= openshift
+# A valid branch or tag from oadp-operator git repo
+OADP_VERSION ?= master
+# namespace to deploy development OADP operator
+OADP_NAMESPACE ?= openshift-adp
+
+.PHONY: deploy-dev
+deploy-dev: TEMP:=$(shell mktemp -d)
+deploy-dev: NAC_PATH:=$(shell pwd)
+deploy-dev: AUX_RANDOM:=$(shell echo $$RANDOM)
+deploy-dev: DEV_IMG?=ttl.sh/oadp-nac-controller-$(shell git rev-parse --short HEAD)-$(AUX_RANDOM):1h
+deploy-dev: ## Build and push development controller image from current branch and deploy development OADP operator using that image
+	IMG=$(DEV_IMG) make docker-build docker-push
+	git clone --depth=1 git@github.com:$(OADP_FORK)/oadp-operator.git -b $(OADP_VERSION)  $(TEMP)/oadp-operator
+	cd $(TEMP)/oadp-operator && \
+	NON_ADMIN_CONTROLLER_PATH=$(NAC_PATH) NON_ADMIN_CONTROLLER_IMG=$(DEV_IMG) OADP_TEST_NAMESPACE=$(OADP_NAMESPACE) \
+		make update-non-admin-manifests deploy-olm && cd -
+	chmod -R 777 $(TEMP) && rm -rf $(TEMP)
```

Comment:
Can we have clean-temp target? It's much safer to check first then chmod and rm:

```Makefile
TEMP_CLEANUP := $(TEMP)/oadp-operator

clean-temp:
	@echo "Cleaning up temporary files..."
	if [ -d "$(TEMP_CLEANUP)" ]; then \
		chmod -R u+w "$(TEMP_CLEANUP)" && rm -rf "$(TEMP_CLEANUP)"; \
			echo "Temporary files cleaned up."; \
	else echo "Temporary directory not found. No cleanup needed."; fi
```

Reply:
Have a PR for this purpose https://github.com/openshift/oadp-operator/pull/1382

Reply:
remove this part of the code, resolving